[BUG] Calendar view is missing

## Description

The calendar view is missing because of the problem with determination of the document type.

DFG Viewer display the documents which are not necessary saved in the database. The current implementation of  `Kitodo\Dlf\ExpressionLanguage\DocumentTypeFunctionProvider` demands that each document is queried from database.

Log entry:
`[ERROR] request="4083a4651344e" component="TYPO3.CMS.Frontend.Configuration.TypoScript.ConditionMatching.ConditionMatcher": Typed property Kitodo\Dlf\ExpressionLanguage\DocumentTypeFunctionProvider::$document must be an instance of Kitodo\Dlf\Domain\Model\Document, null used- TypeError: Typed property Kitodo\Dlf\ExpressionLanguage\DocumentTypeFunctionProvider::$document must be an instance of Kitodo\Dlf\Domain\Model\Document, null used, in file /var/www/extensions/kitodo-presentation/Classes/ExpressionLanguage/DocumentTypeFunctionProvider.php:188 - {"expression":"getDocumentType(3) == \"issue\"","exception":"TypeError: Typed property Kitodo\\Dlf\\ExpressionLanguage\\DocumentTypeFunctionProvider::$document must be an instance of Kitodo\\Dlf\\Domain\\Model\\Document, null used in /var/www/extensions/kitodo-presentation/Classes/ExpressionLanguage/DocumentTypeFunctionProvider.php:188`

## Reproduction

Steps to reproduce the behaviour:

1. Go to [DFG Viewer test system](https://test.dfg-viewer.de/viewer?tx_dlf%5Bid%5D=https%3A%2F%2Fdigital.slub-dresden.de%2Fdata%2Fkitodo%2FTheDarea_416971482-1906%2FTheDarea_416971482-1906_mets.xml)
2. See error

## Expected Behavior

The calendar view should be displayed.

## Screenshots and Examples

At the top expected view and at the bottom current view.

<img width="603" alt="dfgviewer" src="https://github.com/user-attachments/assets/eb514533-134c-492d-9908-f907e78479fc">

--- a/Classes/ExpressionLanguage/DocumentTypeFunctionProvider.php
+++ b/Classes/ExpressionLanguage/DocumentTypeFunctionProvider.php
@@ -54,10 +54,10 @@ class DocumentTypeFunctionProvider implements ExpressionFunctionProviderInterfac
     /**
      * This holds the current document
      *
-     * @var Document
+     * @var Document|null
      * @access protected
      */
-    protected Document $document;
+    protected ?Document $document;
 
     /**
      * @var ConfigurationManager

